pennylane/devices/default_qubit_jax.py

```diff
@@ -0,0 +1,234 @@
+                @qml.qnode(dev, interface="jax", diff_method="backprop")
+                def circuit():
+                    qml.Hadamard(0)
+                    return qml.sample(qml.PauliZ(wires=0))
+                return circuit()
+
+            key1 = jax.random.PRNGKey(0)
+            key2 = jax.random.PRNGKey(1)
+            a = keyed_circuit(key1)
+            b = keyed_circuit(key2) # b will be different samples now.
+
+        Checkout out the `Jax random documentation <https://jax.readthedocs.io/en/latest/jax.random.html>`_
+        for more information.
+
+    Args:
+        wires (int): The number of wires to initialize the device with.
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to 1000 if not specified.
+            If ``analytic == True``, then the number of shots is ignored
+            in the calculation of expectation values and variances, and only controls the number
+            of samples returned by ``sample``.
+        analytic (bool): Indicates if the device should calculate expectations
+            and variances analytically. In non-analytic mode, the ``diff_method="backprop"``
+            QNode differentiation method is not supported and it is recommended to consider
+            switching device to ``default.qubit`` and using ``diff_method="parameter-shift"``.
+        prng_key (jax.random.PRNGKey): The pseudo
+    """
+
+    name = "Default qubit (jax) PennyLane plugin"
+    short_name = "default.qubit.jax"
+
+    parametric_ops = {
+        "PhaseShift": jax_ops.PhaseShift,
+        "RX": jax_ops.RX,
+        "RY": jax_ops.RY,
+        "RZ": jax_ops.RZ,
+        "Rot": jax_ops.Rot,
+        "CRX": jax_ops.CRX,
+        "CRY": jax_ops.CRY,
+        "CRZ": jax_ops.CRZ,
+        "MultiRZ": jax_ops.MultiRZ,
+    }
+
+    C_DTYPE = jnp.complex64
+    R_DTYPE = jnp.float32
+    _asarray = staticmethod(jnp.array)
+    _dot = staticmethod(jnp.dot)
+    _abs = staticmethod(jnp.abs)
+    _reduce_sum = staticmethod(lambda array, axes: jnp.sum(array, axis=tuple(axes)))
+    _reshape = staticmethod(jnp.reshape)
+    _flatten = staticmethod(lambda array: array.ravel())
+    _gather = staticmethod(lambda array, indices: array[indices])
+    _einsum = staticmethod(jnp.einsum)
+    _cast = staticmethod(jnp.array)
+    _transpose = staticmethod(jnp.transpose)
+    _tensordot = staticmethod(
+        lambda a, b, axes: jnp.tensordot(
+            a, b, axes if isinstance(axes, int) else list(map(tuple, axes))
+        )
+    )
+    _conj = staticmethod(jnp.conj)
+    _imag = staticmethod(jnp.imag)
+    _roll = staticmethod(jnp.roll)
+    _stack = staticmethod(jnp.stack)
+
+    def __init__(self, wires, *, shots=1000, analytic=True, prng_key=None):
+        super().__init__(wires, shots=shots, analytic=analytic, cache=0)
+
+        # prevent using special apply methods for these gates due to slowdown in jax
+        # implementation

```

Comment:
Interesting, did you do some comparison tests?

Reply:
I think this comment was left over when I copy and pasted from the TF module. However, I would still argue this is likely the case if it was for tensorflow.